### PR TITLE
updated apd POST endpoint

### LINKED ISSUE #3108: [Dev] Update APD post endpoint to incorporate Mongo
Update the post endpoint, and all of the associated files, to use Mongo instead of Postgres.

### This task is done when…
 - [ ] the post endpoint has been updated to create APDs in Mongo
 - [ ] the tests are updated

### PR DESCRIPTION
**Pull request by thetif**
resolves #3108 

**Description-**

**This pull request changes...**

- updated the connection to mongoose
- updated the createAPD method
- updated the apd POST endpoint
- added tests

**This pull request was tested in the follow ways…**

- run the tests locally
- running endpoint tests
- post to /apds using postman

**Steps to manually verify this change...**

- this is still not working on the PR, so it must be tested locally

1. set the following environment variables
	`export MONGO_URL=mongodb://mongo:cms@localhost:27017/eapd`
	`export MONGO_DATABASE=eapd`

2. run tests in api using `npm test`
3. run endpoint tests `./endpoint-tests/endpoint.sh`
4. use postman to post valid to /apds

### This pull request is ready to review when...

- [X] Automated tests are updated (and all tests are passing)
- [X] The change has been documented
- [X] Changelog is updated as appropriate

### This pull request can be merged when…

- [ ] Code has been reviewed by someone other than the original author
- [ ] QA has verified the accessibility and functionality related to the change
- [ ] Design has approved the experience
- [ ] Product has approved the experience

### CODE REVIEW

**Comment by tbolt** [comment]
I'm seeing this test failing locally-
```
1) api.test.js express api error handling GET /states without a database connection returns 500 HTTP status set to 500:
Error: HTTP status set to 500
+ expected - actual
-500
+400
```
This is with a "fresh" docker build and before running any migrations.

File: api/db/mongodb.js [hunk 1]
```diff
@@ -1,32 +1,41 @@
...
+    return mongoose
+      .connect(connectionString, {
+        useNewUrlParser: true,
+        useUnifiedTopology: true,
+        dbName,
+        authSource: dbName
+      })
+      .then(() => {
+        logger.verbose('MongoDB connected');
+        require('../models'); // eslint-disable-line global-require
```
Comment (knollfear) [inline api/db/mongodb.js:20]: Why can't this be at the top of the file?

**Review by knollfear — COMMENTED** [review]
Tests are all passing locally.

File: api/db/apds.js [hunk 2]
```diff
@@ -1,17 +1,21 @@
...
-  return ids[0];
+const createAPD = async (apd, { APD = mongoose.model('APD') } = {}) => {
+  let newApd = new APD(apd);
+
+  newApd = await newApd.save();
+  return newApd._id; // eslint-disable-line no-underscore-dangle
+};
+
+const deleteAPD = async (id, { APD = mongoose.model('APD') } = {}) => {
+  const found = await APD.findOne({ _id: id });
```
[the remark this answers is not in the document]
Comment (thetif) [reply api/db/apds.js]: I changed it to https://mongoosejs.com/docs/api.html#model_Model.deleteOne because I think findOneAndDelete is being deprecated.

**Comment by thetif** [comment]
@knollfear I think I fixed all of your concerns. Can you take a look again?

**Review by knollfear — APPROVED** [review]
Thanks for making those changes.  LGTM!

**Comment by thetif** [comment]
@tbolt I'm not seeing that failure locally. Are you still getting it?

**Pull request merged**